core/libdeno.rs

```diff
@@ -80,35 +80,98 @@ impl Deref for deno_buf {
   }
 }
 
-impl DerefMut for deno_buf {
+impl AsRef<[u8]> for deno_buf {
   #[inline]
+  fn as_ref(&self) -> &[u8] {
+    &*self
+  }
+}
+
+#[repr(C)]
+pub struct PinnedBufRaw {
+  data_ptr: *mut u8,
+  data_len: usize,
+  pin: PinnedBufPinRaw,
+}
+
+#[repr(C)]
+pub struct PinnedBufPinRaw {
+  ptr: *mut c_void,
+}
+
+#[repr(C)]
+pub struct PinnedBuf {
```

Comment:
Since this is exposed to the user api in crates.io, I think it should have some triple slash explanation.

Reply:
Is `repr(C)` necessary here?

Reply:
It is now, after the latest round of changes.